src/lib.rs

```diff
@@ -45,36 +45,48 @@ struct RustNotify {
     watcher: WatcherEnum,
 }
 
-fn map_watch_error(error: notify::Error) -> PyErr {
+fn map_watch_error(error: notify::Error, ignore_permission_denied: bool) -> Option<PyErr> {
```

Comment:
surely better to leave this function unchanged, and just decide what to do with the error where the function is called?